### ISSUE TITLE
reading bdf properties without bdf

### ISSUE THREAD

**Issue by caseyjlaw** [demorest/sdmpy#8]
I sometimes use sdmpy to browse metadata that is related to a bdf (e.g., the bdf file name). However, the current set up requires that the bdf be present to view this info. So this errors out:
```
> sdm = sdmpy.SDM(sdmname)
> scan = sdm.scan(1)
> print(scan.bdf.fname)
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-20-ac34a58b1246> in <module>()
----> 1 print(scan.bdf.fname)

/home/cbe-master/realfast/anaconda/lib/python2.7/site-packages/sdmpy/scan.pyc in bdf(self)
     44     def bdf(self):
     45         if self._bdf is None:
---> 46             self._bdf = BDF(self._bdf_fname)
     47         return self._bdf
     48 

/home/cbe-master/realfast/anaconda/lib/python2.7/site-packages/sdmpy/bdf.pyc in __init__(self, fname)
     65     def __init__(self, fname):
     66         self.fname = fname
---> 67         self.fp = open(fname, 'r')
     68         self.mmdata = mmap.mmap(self.fp.fileno(), 0, mmap.MAP_PRIVATE,
     69                 mmap.PROT_READ)

IOError: [Errno 2] No such file or directory: '/lustre/evla/wcbe/data/no_archive/uid___evla_bdf_X1'

```

I see that the hidden variable `_bdf_fname` holds that particular name, but I wondered if there was a way to make some metadata available without actually trying to read the bdf.

**Comment by demorest** [comment]
Yes, this is a good suggestion.  I've been sometimes needing to use the _bdf_fname variable also which is probably a good sign that it should be "officially" exposed.  I just added "bdfdir" and "bdf_fname" properties to Scan that give the directory and full path to the BDF, respectively.

Can you think of any other metadata besides the filename that would be helpful to include here?

**Comment by demorest** [comment]
While we're at it, I've been thinking about putting in a test for the special "X1" missing-BDF filename.  Do you think having scan.bdf return None in this case would be helpful?  Or would that just be more confusing, since then you will get cryptic NoneType-related exceptions rather than IOError when trying to do something with the BDF?

**Comment by caseyjlaw** [comment]
Great!
Returning None would be more Pythonic. I can see how IOError is more clear though. Perhaps the latter could be avoided by properly testing for the presence of a bdf before reading?

**Comment by demorest** [comment]
Addressed via PR #9
